.github/workflows/ci.yaml

```diff
@@ -0,0 +1,50 @@
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+permissions: read-all
+
+jobs:
+  test-architectures:
+    name: test OS/arch combinations
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - 'ubuntu-24.04'
+          - 'ubuntu-24.04-arm'
+          - 'macos-13' # the last intel mac
+          - 'macos-15' # arm64 mac
+          - 'windows-2025'
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: install local package
+        run: |
+          python -m pip install .
+
+      - name: test that editorconfig-checker works by letting it output it's version
+        run: |
+          ec --version
```

Comment:
This is also a kinda weak test, as there's only the exit code that can make the step, thus the workflow, fail.

I remember that for Megalinter, the published binaries for a go tool (probably revive), didn't actually have a version information (maybe limited to the ones released through Docker images). The workaround needed was to build ourselves, like

https://github.com/oxsecurity/megalinter/blob/73c05baa4078059527411e1784a0db4f8b9df94b/megalinter/descriptors/go.megalinter-descriptor.yml#L86-L101

Reply:
I was intentionally mirroring the `make test` - if we decide to improve this we should do it in both places at the same time.